test/parallel/test-repl-custom-eval.js

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+require('../common');
+const ArrayStream = require('../common/arraystream');
+const assert = require('assert');
+const { describe, it } = require('node:test');
+
+const repl = require('repl');
+
+describe('repl with custom eval', () => {
```

Comment:
We can probably run them in parallel when running the file standalone
```suggestion
describe('repl with custom eval', { concurrency: !process.env.TEST_PARALLEL }, () => {
```

Reply:
Sounds good :slightly_smiling_face: 

But is `{ concurrency: !process.env.TEST_PARALLEL }` correct?

a falsy concurrency value means that the tests won't be run in parallel right? so we are saying that if `TEST_PARALLEL` is truthy then the tests here need to be run sequentially? or am I misunderstanding?

Reply:
`TEST_PARALLEL` is set by Python runner when it's running test in parallel. When it is, we don't want the Node.js one to parallelize on its own, otherwise it could oversubscribe the machine – although I'm not sure if that's the case, I originally thought `getReplOutput` was spawning a subprocess, but if that's not the case, it shouldn't really matter

Reply:
Thanks for the explanation @aduh95 :pray: (it is still a bit murky to me but I do get the gist of it :sweat_smile:)

Regarding `getReplOutput` no, I am quite sure that it doesn't spawn a subprocess, since it simply starts a `REPLServer` which does run in the same process (and it uses `runInContext` and `runInThisContext` to evalutate code)

So given the above, are you happy with the current version of the code? :slightly_smiling_face:

Reply:
Currently it runs the test serially, so no I'm not happy with it unless we have a good reason to do that 😅 Would `concurrency: true` work? If so, we should use it, if not, we should set `concurrency: false` explicitly with a comment explaining why – but please treat this as a nit and feel free to ignore

Reply:
ah okok I see, no I'm totally happy to add `concurrency: true` :slightly_smiling_face::+1:

Reply:
`concurrency: true` added (https://github.com/nodejs/node/pull/57691/commits/63034cb9808b0abbb564ee856dedeba828b9a096) :slightly_smiling_face: 

Also `concurrency: true`  was actually causing some failures because the fact that I was using the same global variable `foo` in two different tests so I fixed that, this also helped me notice that the `useGlobale: false` test could be improved to make sure global variables are inherited by the REPL (which is a documented behavior: https://nodejs.org/api/repl.html#global-and-local-scope), thanks for that! :smile: :heart_hands: 

Please have a look and let me know if things look good to you now